### PR TITLE
Dynamic generation of the CI test matrix

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR adds dynamic generation of the CI test matrix based on contents of `src/integration/stimulus/L0_regression.yml`. This removes the need of keeping duplicated list of tests in the CI workflow file. Also with this PR all the regression tests are run in parallel in separate jobs which should speed up the CI execution

### CODE REVIEW

**Comment by howardtr** [comment]
The changes themselves lgtm 

Please wait for @anjpar / @calebofearth to review as well, thanks!

(I suspect that we'll need to update the branch rules @anjpar @calebofearth @korran)

**Review by howardtr — APPROVED** [review]

File: tools/scripts/Makefile
Comment (calebofearth) [inline tools/scripts/Makefile:1]: Switching to rv32imc looks like an appropriate fix per the VeeR PRM, however we inherited this setting directly from the makefile in the VeeR repository:
https://github.com/chipsalliance/Cores-VeeR-EL2/blob/main/tools/Makefile
Are you aware of any discussion around why this config is used over there?

**Review by calebofearth — APPROVED** [review]
Changes all look good, just had a question on the Makefile.
Also, I created dev-public so we can retarget.

File: tools/scripts/Makefile
Comment (mkurc-ant) [reply tools/scripts/Makefile:1]: Hmm, I'm not aware of such a discussion. I had to change to `rv32imc` as there wasn't a build of `libc` for `rv32imac` that comes with the pre-built toolchain used in the CI. This caused some tests to fail on firmware build. Besides that, according to the documentation VeeR is `imc` - does not support atomics.

**Comment by calebofearth** [comment]
> @calebofearth Its retargeted. Can we merge it?

I will merge it with a Squash. Thanks.

**Pull request merged**